keps/sig-network/3698-multi-network/README.md

```diff
@@ -0,0 +1,363 @@
+    - [Phase III (basic Kubernetes features integration)](#phase-iii-basic-kubernetes-features-integration)
+    - [Phase IV (extended functionality)](#phase-iv-extended-functionality)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Graduation Criteria](#graduation-criteria)
+    - [Alpha](#alpha)
+    - [Beta](#beta)
+- [Production Readiness Review Questionnaire](#production-readiness-review-questionnaire)
+  - [Feature Enablement and Rollback](#feature-enablement-and-rollback)
+  - [Rollout, Upgrade and Rollback Planning](#rollout-upgrade-and-rollback-planning)
+  - [Monitoring Requirements](#monitoring-requirements)
+  - [Dependencies](#dependencies)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+One of the main requirements for Kubernetes networking today is to enable
+connectivity between `Pods` in a cluster. This facility satisfies a large number
+of Kubernetes users, but it isn't sufficient for some important cases:
+
+* applications leveraging different isolated networks exposed through different
+  interfaces
+* applications leveraging performance-oriented interfaces (e.g. `AF_XDP`,
+  `memif`, `SR-IOV`)
+* applications requiring support for protocols not yet supported by Kubernetes
+
+Here we propose a solution that enables multiple network interfaces to be
+defined in the `Pod` specification to provide a standard and clear mechanism to
+implement these more complex configurations.
+
+## Motivation
+
+We want to have a common API allowing us to define a catalog of different
+networks in the Kubernetes cluster. It would allow attaching a pod to one or
+several networks via a given type of interface depending on its connectivity
+or performance needs.
+
+### Goals
+
+* Define user stories and requirements for the Multi-Network effort in Kubernetes.
+* Introduce API object to kubernetes describing networks Pod can attach to.
+* Evolve current Kubernetes networking model to support multiple networks,
+defining the new model in a backwards compatible way.
+* Integrate this API with the existing core objects.
+* Define “reference implementation” that can be used in Kubernetes CI.
+* Define conformance tests.
+
+
+### Non-Goals
+
+Define the CNI implementation.
+
+## Proposal
+
+### Personas
+
+#### Cluster operator
+
+This role will be responsible for defining and managing PodNetworks (see below)
+that properly describe the infrastructure available for the cluster, namespace
+and workloads. This persona can define which users can “attach” to a specific
+PodNetwork.
+
+#### Application developer
+
+**Application developer** is the consumer of PodNetwork via referencing them in
+their workloads. Application developers usually will not create or remove the
+PodNetwork on their own.
```

Comment:
I'm not sure this is the target of a low level primitive like the network , cc: @smarterclayton @thockin

Reply:
my question is why an application developer should care about referencing on or other network, what kind of applications/workloads  should have the notion of Network?

Reply:
partially you are right, story 6 is about providing PodNetwork profile per ns, so that application dev did not care about this, but in other cases application dev is the one that will care which networks to attach to.

I think you are trying to make this KEP to be applicable to 100% of k8s userbase, which it is not and never will be. Today it is applicable to maybe 10-20%, and in future we can maybe get to 30-40%, is this not acceptable ?